src/main.cpp

```diff
@@ -7458,8 +7520,12 @@ bool SendMessages(const Consensus::Params& params, CNode* pto)
 
                 for (const auto& txinfo : vtxinfo) {
                     const uint256& hash = txinfo.tx->GetHash();
-                    CInv inv(MSG_TX, hash);
+                    CInv inv = InvForTransaction(txinfo.tx);
                     pto->setInventoryTxToSend.erase(hash);
+                    // ZIP 239: We won't have v5 transactions in our mempool until after
+                    // NU5 activates, at which point we will only be connected to peers
+                    // that understand MSG_WTX.
+                    if (inv.type == MSG_WTX) assert(pto->nVersion >= CINV_WTX_VERSION);
```

Comment:
Can you go over why it is correct for this to be an assertion (rather than log-and-ignore)?

Reply:
Ah, these assertions slipped into the commits here, but should have been part of a later commit (that alters the `CInv` serialization to make it an error to use `MSG_WTX` in the current network, as I believe we'd discussed implementing). This is the extra piece I was having trouble getting to behave correctly.